test-specifications/utility.md

```diff
@@ -0,0 +1,293 @@
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "method": "generateKey"
+}
+```
+
+*Generates an ED25519 private key*
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 2,
+  "method": "generateKey",
+  "params": {
+    "type": "ed25519PrivateKey"
+  }
+}
+```
+
+*Generates an ED25519 public key*
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 3,
+  "method": "generateKey",
+  "params": {
+    "type": "ed25519PublicKey"
+  }
+}
+```
+
+*Generates the serialized protobuf ECDSAsecp256k1 public key that is paired with the input ECDSAsecp256k1 private key*
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 4,
+  "method": "generateKey",
+  "params": {
+    "type": "ecdsaSecp256k1PublicKey",
+    "privateKey": "3030020100300706052b8104000a04220420e8f32e723decf4051aefac8e2c93c9c5b214313817cdb01a1494b917c8436b35",
+    "protobufBytes": true
+  }
+}
+```
+
+*Generates a threshold key that requires two keys to sign, and contains a random key, an ED25519 private key, and an ECDSAsecp256k1 public key that is paired with the input ECDSAsecp256k1 private key*
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 5,
+  "method": "generateKey",
+  "params": {
+    "type": "thresholdKey",
+    "threshold": 2,
+    "keys": [
+      {},
+      {
+        "type": "ed25519PrivateKey"
+      },
+      {
+        "type": "ecdsaSecp256k1PublicKey",
+        "privateKey": "3030020100300706052b8104000a04220420e8f32e723decf4051aefac8e2c93c9c5b214313817cdb01a1494b917c8436b35"
+      }
+    ]
+  }
+}
+```
+
+*Generate a key list that contains two key lists. The first key list contains an ED25519 private key, a random key, and an ECDSAsecp256k1 public key that is paired with the input ECDSAsecp256k1 private key. The second key list contains an ECDSAsecp256k1 private key, a threshold key, and a random key. The threshold key requires two keys to sign, and contains two random keys and an ED25519 public key that is paired with the input ED25519 private key.*
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 6,
+  "method": "generateKey",
+  "params": {
+    "type": "keyList",
+    "keys": [
+      {
+        "type": "keyList",
+        "keys": [
+          {
+            "type": "ed25519PrivateKey"
+          },
+          {},
+          {
+            "type": "ecdsaSecp256k1PublicKey",
+            "privateKey": "3030020100300706052b8104000a04220420e8f32e723decf4051aefac8e2c93c9c5b214313817cdb01a1494b917c8436b35"
+          }
+        ]
+      },
+      {
+        "type": "keyList",
+        "keys": [
+          {
+            "type": "ecdsaSecp256k1PrivateKey"
+          },
+          {
+            "type": "thresholdKey",
+            "threshold": 2,
+            "keys": [
+              {},
+              {},
+              {
+                "type": "ed25519PublicKey",
+                "privateKey": "302e020100300506032b657004220420c036915d924e5b517fae86ce34d8c76005cb5099798a37a137831ff5e3dc0622"
+              }
+            ]
+          },
+          {}
+        ]
+      }
+    ]
+  }
+}
+```
+
+#### JSON Response Examples
```

Comment:
Maybe it would be good to put response example under the approp. request example.

Reply:
I like that, makes it easier to follow. Fixed